misc/bazel/pkg.bzl

```diff
@@ -474,6 +475,8 @@ def codeql_pack(
         visibility = ["//visibility:public"],
     )
     _codeql_pack_install(internal("installer"), [name], install_dest = install_dest, apply_pack_prefix = False)
+    if not native.existing_rule("install"):
```

Comment:
I don't like this implementation - per https://bazel.build/rules/lib/toplevel/native#existing_rule, this API will be deprecated and removed at some point:
> If possible, use this function only in [implementation functions of rule finalizer symbolic macros](https://bazel.build/extending/macros#finalizers). Use of this function in other contexts is not recommened, and will be disabled in a future Bazel release; it makes BUILD files brittle and order-dependent.


Given the fairly small impact of this, I'm fine merging this now, but we will have to remove this feature again once bazel follows through with the removal of this function.

Reply:
good point! I wasn't even aware of that (or of the new macro definition framework)

I agree it's better to avoid that, and have changed the code to do that, and added an `experimental` flag to aid the definition of experimental packs. Alas, I still needed `existing_rule` because the internal repo defines two `codeql_pack`s in a bazel package (the java single and multi kotlin variants). So for backward compatibility I've left that. Once the internal repo is updated (probably we can put `install_dest = None` on all `codeql_packs` there), we can remove `TODO`s here with a third PR.